extensions/warp-ipfs/src/thumbnail.rs

```diff
@@ -44,13 +46,23 @@ impl Default for ThumbnailId {
     }
 }
 
-#[derive(Default, Clone)]
+#[derive(Clone)]
 #[allow(clippy::type_complexity)]
 pub struct ThumbnailGenerator {
-    task: Arc<Mutex<BTreeMap<ThumbnailId, JoinHandle<Result<(ExtensionType, Vec<u8>), Error>>>>>,
+    ipfs: Ipfs,
+    task: Arc<
```

Comment:
the name is misleading IMO. Should be either `tasks` (plural) or `task_handles`.